jenkins-integration/jenkins-jobs/common/scripts/job_steps/040_install_puppet_code.sh

```diff
@@ -1,20 +1,24 @@
 #!/bin/bash
 
+pushd jenkins-integration
+source jenkins-jobs/common/scripts/job_steps/initialize_ruby_env.sh
+
 # This job sets up the following:
-# - PE 2015.3.1 installation on provided master
 # - OPS environments installed on master
 # - CatalogZero module added to production environment
-# - Node classified via NC to have catalog zero module
 #
 # Note that we only need the OPS environments installed, we don't need to
 # explicitly reference them or use any of their classes; simply having them
 # there will exhibit the slow performance behavior we're trying to highlight
 # with this job.
 
+# TODO TODO TODO: this should *NOT* be hard-coded in the job template,
+#  this is very specific to a given job.
```

Comment:
Referring to this file, really.  The current implementation of deploying the code is doing a very special / unusual thing that downloads an old snapshot of the ops modules from a tarball, because it can take over an hour to deploy it from r10k.  Ultimately this file needs to be generalized to support some more standard methods of deploying code.  I expect we'll need to do that in one of the next few tickets.